### DISCUSSION TITLE
Custom fieldtype label showing in CP, but not the component itself.

### DISCUSSION THREAD

**Discussion by anakadote** [statamic/cms#8518] [Troubleshooting] [answered]
I just upgraded a Statamic v3.4 website to v4.14 (fwiw) and am now trying to create a custom fieldtype. I have followed [the instructions](https://statamic.dev/extending/fieldtypes) but have an issue where my custom fieldtype is displaying in the CP, but only the label. I have ensured that my Vue component has content to display. I can see in Chrome devtools that the component is hidden (it displays as `<!---->`)

Is there something obvious that I'm missing?

<img width="516" alt="Screen Shot 2023-07-31 at 7 11 28 PM" src="https://github.com/statamic/cms/assets/2798169/9c5dcfdf-6c4b-49d2-a52b-928201720e34">
<img width="444" alt="Screen Shot 2023-07-31 at 7 20 58 PM" src="https://github.com/statamic/cms/assets/2798169/fa2e3165-c4d3-4ac3-99bb-7a16ee0504df">
<img width="866" alt="Screen Shot 2023-07-31 at 7 13 03 PM" src="https://github.com/statamic/cms/assets/2798169/4bbc27a8-1256-4883-9946-e633917888a4">

**Comment by duncanmcclean** [comment]
Are you running your addon's watcher (`npm run dev` or `npm run build`) so changes to your Vue components get recompiled?

**Comment by anakadote** [comment]
@duncanmcclean yep, and I can even run code in the mounted() hook:
```
<template>
    <div>
        Hello World
    </div>
</template>

<script>
export default {
    mixins: [Fieldtype],
    data() {
        return {}
    },
    mounted() {
        console.log('hello, world')
    }
};
</script>
```
<img width="392" alt="Screen Shot 2023-08-01 at 8 18 26 AM" src="https://github.com/statamic/cms/assets/2798169/15cb98dc-8bab-49c6-8a07-3f5791db5dc6">

**Comment by anakadote** [comment, marked as answer]
I tried a bunch of different things to no avail, but was finally able to get my component to show up by inlining in:
```
<script>
export default {
    mixins: [Fieldtype],
    template: `<div>Hello World</div>`
};
</script>
```

**Comment by laborb-hs** [comment]
I recently had the same problem in an older statamic project.
It was still working with webpack and the project was upgraded to vue 3.x at some time.
So the CP fieldtype was not properly compiled for use in the still-vue-v2 control panel.

The solution was to upgrade the project to vite and then separate the compiling vue2 for cp and vue3 for frontend, as the documentation suggests.

In my case the inline templating was not resolving the issue.

Link to the docs: https://statamic.dev/extending/control-panel#using-vite
